artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompConnection.java

```diff
@@ -566,7 +566,7 @@ StompPostReceiptFunction subscribe(String destination,
          if (selector == null) {
             selector = noLocalFilter;
          } else {
-            selector += " AND " + noLocalFilter;
+            selector = "(" + selector + ") AND " + noLocalFilter;
```

Comment:
could selector be an empty string?

Reply:
Yes, I believe so. I'll add some validation.

Reply:
Validation added.